client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c

```diff
@@ -1006,23 +1008,29 @@ void kaa_tcp_channel_kaasync_message_callback(void *context, kaatcp_kaasync_t *m
     KAA_RETURN_IF_NIL2(context, message, );
     kaa_tcp_channel_t *channel = (kaa_tcp_channel_t *) context;
 
-    KAA_LOG_TRACE(channel->logger, KAA_ERR_NONE, "Kaa TCP channel [0x%08X] KAASYNC message received"
-                                                                            , channel->access_point.id);
+    KAA_LOG_TRACE(channel->logger, KAA_ERR_NONE, "Kaa TCP channel [0x%08X] KAASYNC message received",
+                  channel->access_point.id);
 
     uint8_t zipped = message->sync_header.flags & KAA_SYNC_ZIPPED_BIT;
     uint8_t encrypted = message->sync_header.flags & KAA_SYNC_ENCRYPTED_BIT;
 
-    if (!zipped && !encrypted) {
-        kaa_error_t error_code = kaa_platform_protocol_process_server_sync(
-                        channel->transport_context.kaa_context->platform_protocol,
-                        (const uint8_t *)message->sync_request, message->sync_request_size);
-        if (error_code) {
-            KAA_LOG_ERROR(channel->logger, error_code, "Kaa TCP channel [0x%08X] failed to process server sync"
-                                                                                    , channel->access_point.id);
-        }
+#ifdef KAA_ENCRYPTION
+    if (encrypted) {
+        ext_decrypt_data((uint8_t *)message->sync_request, message->sync_request_size,
+                         (uint8_t *)message->sync_request, &message->sync_request_size);
+    }
+#endif
+    /* Currently not supported */
+    if (!zipped) {
+        kaa_error_t error_code = kaa_platform_protocol_process_server_sync(channel->transport_context.kaa_context->platform_protocol,
+                                                                           (const uint8_t *)message->sync_request,
+                                                                           message->sync_request_size);
+        if (error_code)
+            KAA_LOG_ERROR(channel->logger, error_code, "Kaa TCP channel [0x%08X] failed to process server sync",
+                          channel->access_point.id);
     } else {
-        KAA_LOG_DEBUG(channel->logger, KAA_ERR_NONE, "Kaa TCP channel [0x%08X] received unsupported flags: zipped %d, encrypted %d"
-                                                                                        , channel->access_point.id, zipped, encrypted);
+        KAA_LOG_DEBUG(channel->logger, KAA_ERR_NONE, "Kaa TCP channel [0x%08X] received unsupported flags: zipped %d, encrypted %d",
```

Comment:
We couldn't process sync here. I think the log level should be higher than `DEBUG`, e.g. `WARN`.